driver/src/price_finding/naive_solver.rs

```diff
@@ -164,7 +180,11 @@ impl PriceFinding for NaiveSolver {
 
             // apply fee to volumes account for rounding errors, moving them to
             // the fee token
-            for (i, order) in orders.iter().enumerate() {
+            let (order_1_index, order_2_index) =
+                matching_orders_indexes.expect("fee price was nonzero so we should have a match");
+            for i in &[order_1_index, order_2_index] {
+                let i = *i;
+                let order = &orders[i];
```

Comment:
Is there something wrong with dereferencing directly inside the array access? It doesn't appear to be used more than once.

Reply:
Nothing wrong with that but the `i` is used more than once so we would have `*i` a bunch of times and I wanted to keep the diff cleaner.

Reply:
Oh, sorry, I did no see the other usages! Cool stuff!